### PR TITLE
editing iterables part

### PR DESCRIPTION
**Pull request by djarecka**
@miykael - I edited iterables part, wanted to emphasize that it had to be used with Workflow (it wasn't clear to me at the beginning), so changed the part with IdentityInterface. Feel free to edit my changes.

@satra - let me know if you have more suggestions

### CODE REVIEW

File: notebooks/basic_iteration.ipynb [hunk 1]
```diff
@@ -157,9 +159,10 @@
    "source": [
     "# ``IdentityInterface`` (special use case of ``iterabels``)\n",
```
Comment (satra) [inline notebooks/basic_iteration.ipynb]: Iterabels --> iterables

**Comment by satra** [comment]
@djarecka - i think we should move over the content from the nipype docs into the tutorial. i would like replace any user docs through this tutorial.

File: notebooks/basic_iteration.ipynb (hunk 1 above)
Comment (djarecka) [reply notebooks/basic_iteration.ipynb]: thanks

**Comment by satra** [comment]
@djarecka - yes, copy and paste descriptions into the notebooks and remove the description from the nipype documentation. essentially i would like to see literate documentation, i.e. documentation that can be executed and tested. this doesn't have to happen for this PR.

**Comment by djarecka** [comment]
@satra ok, I can start moving documentation to notebook, but I'd suggest that this should be independent on the tutorial and kept somewhere else. My understanding is that, even if the tutorial covers many things from the documentation, it should have a different structure and flow, and putting documentation notebooks together with tutorial would generate a bit of chaos, but you or @miykael could have different view.

what about keeping the documentation within nipype repo? or creating a new repo?

**Comment by satra** [comment]
@djarecka - all the documentation is not moving. just all tutorials and user docs that are better served as tutorials.

**Comment by djarecka** [comment]
@satra - hmm, I'm still not sure about it.. I believe, that too many notebooks, that partially cover the same materials doesn't help, especially if you're starting learning a new tool. Maybe we can at least create an "advanced part" and put everything there so beginners don't go there right away, and it's clear that it is somehow independent set? 

Anyway, I'll let you and @miykael decide. Right now, I want to finish exercises for the current notebooks anyway.

**Comment by miykael** [comment]
@djarecka, the changes look good and I like the exercise. But I'm not yet fully sure how to have this in a nice way for the homepage. So that the solution is accessible, but hidden in a dropdown or something similar.

**Comment by miykael** [comment]
@satra, that's a good idea and I'm very happy if we can move the other tutorials and the user documentation to this tutorial.

To make sure that we don't do too much/little, the idea is to move:
- all the user documents (http://nipype.readthedocs.io/en/latest/users/index.html)
- all workflow examples (and make them executable with example data, using datalad for example)

Would you also want us to move the developer's guide (http://nipype.readthedocs.io/en/latest/devel/index.html)?

@djarecka, how do you want to handle this? I'm happy to dig my teeth into this, but I don't want to interfere with what you've already done. How do you want to split this?

**Comment by miykael** [comment]
@djarecka, I've changed the layout of the iterables exercise to this: https://miykael.github.io/nipype_tutorial/notebooks/basic_iteration.html#Exercise-1. Like this, the solution is hidden when the notebook is run locally and only shown once on the homepage. What do you think?

**Comment by satra** [comment]
@miykael: let's start with the user documents first. i think the workflow examples can be tested in other ways (and many of them are long running examples), so we should wait on that for now.

regarding the exercise, there is an option in jupyter to embed the widgets in a notebook which should then behave the same way on the html page.

**Comment by djarecka** [comment]
@miykael : i haven't started working on it. I guess we don't have to do in one PR. what about you'll start and suggest the new structure of the directories and I can later continue with other pages from User Guide.
I would only ask to not change the `basic_*` notebook before accepting my PR with exercises (have some problem with `SelectFiles`, but once i know the answer, I can finish it).

**Comment by miykael** [comment]
@satra, I looked into how to embed jupyter widgets into html (http://ipywidgets.readthedocs.io/en/stable/embedding.html) but was unable to do it with the exercise nbextension. Do you have another pointer/example? @djarecka, can you look into this, perhaps you have more luck.

**Pull request merged**